Cython/Compiler/Main.py

```diff
@@ -248,9 +248,12 @@ def find_module(self, module_name, from_module=None, pos=None, need_pxd=1,
                     # Set pxd_file_loaded such that we don't need to
                     # look for the non-existing pxd file next time.
                     scope.pxd_file_loaded = True
-                    package_pathname = self.search_include_directories(
-                        qualified_name, suffix=".py", source_pos=pos)
-                    if package_pathname and package_pathname.endswith(Utils.PACKAGE_FILES):
+                    if any(
+                        Utils.is_package_file(
+                            self.search_include_directories(
+                                qualified_name, suffix=s, source_pos=pos)
+                        ) for s in [".py", ".so", ".pyd"]
+                    ):
```

Comment:
This is the wrong way around. We should search the PYTHONPATH in order, not three times for different files. Otherwise, we'd end up picking a late `.py` file over an earlier `.so` file, and use the wrong package directory.